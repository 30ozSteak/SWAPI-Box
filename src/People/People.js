import React, { Component } from "react";
import PeopleCard from "../PeopleCard/PeopleCard";
import Loading from "../Loading/Loading";
import "../PeopleCard/PeopleCard.css";

const People = ({ people, planets, species, handlePeopleLink }) => {
  try {
    const characters = people.results.map(person => (
      <PeopleCard people={people} species={species} planets={planets} />
    ));
    let charactersToRender = characters.slice(0, 1);
    return (
      <div className="container-header">
        <h5>Hover Over a Character to Learn More</h5>
        <div className="people-container">{charactersToRender}</div>;
      </div>
    );
  } catch {
    return (
      <div>
        <h1>
          <Loading />
        </h1>
      </div>
    );
  }
};
export default People;