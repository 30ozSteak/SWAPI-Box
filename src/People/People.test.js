import React from "react";
import People from "./People";
import { shallow } from "enzyme";

describe("People", () => {

  it("should match the screenshot", () => {
    const mockPeople = ['results': [{people: "Luke"}, {people: 'Darth Vador'}]]

    const mockPlanets = [{homeworld: "Moon"}]
    const mockSpecies = [{species: "driod"}]
    
    const wrapper = shallow(<People people={mockPeople} planets={mockPlanets} species={mockSpecies} />);

    expect(wrapper).toMatchSnapshot();
  });
})
