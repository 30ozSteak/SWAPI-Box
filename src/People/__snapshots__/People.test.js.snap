// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People should match the screenshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <People
    people={
      Array [
        "results",
      ]
    }
    planets={
      Array [
        Object {
          "homeworld": "Moon",
        },
      ]
    }
    species={
      Array [
        Object {
          "species": "driod",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Loading People
      </h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading People",
      },
      "ref": null,
      "rendered": "Loading People",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Loading People
        </h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading People",
        },
        "ref": null,
        "rendered": "Loading People",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
